op-deployer/pkg/deployer/pipeline/alt_da.go

```diff
@@ -0,0 +1,56 @@
+package pipeline
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/opcm"
+	"github.com/ethereum-optimism/optimism/op-deployer/pkg/deployer/state"
+	"github.com/ethereum/go-ethereum/common"
+)
+
+func DeployAltDA(env *Env, intent *state.Intent, st *state.State, chainID common.Hash) error {
+	lgr := env.Logger.New("stage", "deploy-alt-da")
+
+	chainIntent, err := intent.Chain(chainID)
+	if err != nil {
+		return fmt.Errorf("failed to get chain intent: %w", err)
+	}
+
+	chainState, err := st.Chain(chainID)
+	if err != nil {
+		return fmt.Errorf("failed to get chain state: %w", err)
+	}
+
+	if !shouldDeployAltDA(chainIntent, chainState) {
+		lgr.Info("alt-da deployment not needed")
+		return nil
+	}
+
+	var dao opcm.DeployAltDAOutput
+	lgr.Info("deploying alt-da contracts")
+	dao, err = opcm.DeployAltDA(env.L1ScriptHost, opcm.DeployAltDAInput{
+		Salt:                     st.Create2Salt,
+		ProxyAdmin:               st.ImplementationsDeployment.OpcmProxyAddress,
+		ChallengeContractOwner:   chainIntent.Roles.L1ProxyAdminOwner,
+		ChallengeWindow:          new(big.Int).SetUint64(chainIntent.DangerousAltDAConfig.DAChallengeWindow),
+		ResolveWindow:            new(big.Int).SetUint64(chainIntent.DangerousAltDAConfig.DAResolveWindow),
+		BondSize:                 new(big.Int).SetUint64(chainIntent.DangerousAltDAConfig.DABondSize),
+		ResolverRefundPercentage: new(big.Int).SetUint64(chainIntent.DangerousAltDAConfig.DAResolverRefundPercentage),
+	})
+	if err != nil {
+		return fmt.Errorf("failed to deploy alt-da contracts: %w", err)
+	}
+
+	chainState.DataAvailabilityChallengeProxyAddress = dao.DataAvailabilityChallengeProxy
+	chainState.DataAvailabilityChallengeImplAddress = dao.DataAvailabilityChallengeImpl
```

Comment:
UX nit: If we have a chain that is already an alt-da chain but is being used with op-deployer for a reason other than to run the `DeployAltDA` pipeline task, then these addresses won't be added to the `state.json` and the values will be `0x0000...`. Not a pressing issue but something maybe we can keep track off be label as 'good first issue'?

Reply:
Is this causing a problem? Like would you prefer the zero addresses not show up in the state at all?

Reply:
Not causing a problem. I was just referencing this issue: https://github.com/ethereum-optimism/platforms-team/issues/347